src/server/lib/middleware/login.ts

```diff
@@ -114,6 +95,31 @@ export const loginMiddlewareOAuth = async (
 				scopesForApplication.includes(scope as Scopes),
 			)
 		) {
+			// valid tokens at this point
+			// 2. Check if the user has the `GU_SO` cookie
+			if (req.cookies.GU_SO) {
+				// if the user has a `GU_SO` cookie, compare the value to the `iat` claim in the access and id tokens
+				// if the `GU_SO` cookie (lastSignOut) is newer than the tokens, the user has recently signed out and we need to clear the tokens
+				const lastSignOut = parseInt(req.cookies.GU_SO, 10);
+				const accessTokenIat = accessToken.claims.iat!;
```

Comment:
I ended up doing [this](https://github.com/guardian/manage-frontend/pull/1288/files#diff-44ef2ef130a808eb6420fdf16f81e941fb80cc6fa14d697c797c7768c317a0abR131) in MMA to handle the existence of `.claims.iat` - maybe you could do similar here to avoid the non-null assertion?